src/lib/mina_caqti/README.md

```diff
@@ -0,0 +1,76 @@
+Mina_caqti
+==========
+
+This library is designed to assist in querying relational databases
+using the Caqti library. It is used extensively for querying the
+archive database in the modules `Processor` and `Load_data` modules in
+`Archive_lib`.
+
+Constructing SQL queries
+------------------------
+
+Instead of writing out SQL queries as text, the
+functions here can construct those queries from table information.
+
+For example, the `Token` module in the archive processor contains:
+```ocaml
+  let table_name = "tokens"
+
+  let find_by_id (module Conn : CONNECTION) id =
+    Conn.find
+      (Caqti_request.find Caqti_type.int typ
+         (Mina_caqti.select_cols_from_id ~table_name ~cols:Fields.names) )
+      id
+```
+The list `Fields.names` is generated from the `deriving fields` annotation on
+the type `Token.t`. The call to `select_cols_fromid` constructs the query
+```
+SELECT value,owner_public_key_id,owner_token_id FROM tokens WHERE id = ?
+```
+
+There are other SQL-building functions in the library, like
+`select_cols`, `insert_into_cols`, and `select_insert_into_cols`, which
+are documented in the source code.
+
+Custom array types
+------------------
+
+Another notable feature of the library are the custom array types,
+which are used to provide a `Caqti.Type.t` for OCaml array types not
```

Comment:
```suggestion
You can use custom array types to provide a `Caqti.Type.t` for OCaml array types not
```

Reply:
imo, by describing the custom array types here, they are notable so we can just introduce them